kratos/containers/master_slave_relation.h

```diff
@@ -0,0 +1,197 @@
+//    ' /   __| _` | __|  _ \   __|
```

Comment:
this is an implementation detail, so i think it belongs in the same directory as the builderandsolver which uses it. 
(please tell so in the comment describing the object)

Reply:
i think that this one as "implentation detail" of the builder and solver does not belong in this PR

also i would use a name that suggests immediately that this is internal (maybe putting it in a namespace "Internal" or "detail" or something of the sort